base/src/main/java/org/apache/cassandra/auth/LDAPAuthenticator.java

```diff
@@ -136,6 +137,15 @@ public AuthenticatedUser authenticate(String username, String password) throws A
     {
         try
         {
+
+            if (!Boolean.parseBoolean(properties.getProperty(ALLOW_BLANK_PASSWORD_PROP, "true")))
```

Comment:
Could you please create a static method in `LdapAuthenticatorConfiguration` which would have signature `public static boolean parseBoolean(String propertyName, Boolean defaultValue)` ?

Then this would be simplified to `if (parseBoolean(ALLOW_BLANK_PASSWORD_PROP, true)`.

You will also need to change other places where similar parsing is done.

Reply:
Could you also please move the resolution of this to `setup` method? There is no reason to compute this every single time we are going to authenticate a user. It is not like this value would be a subject of change. It is statically set, just once. So just resolve it in `setup` and create a private field in that class to hold that value.

Reply:
changed to use private field allow_empty_password in LDAPAuthenticator class:
`private boolean allow_empty_password;`
fill this field on setup method:
`allow_empty_password = parseBoolean(properties.getProperty(ALLOW_EMPTY_PASSWORD_PROP));`
and replace check password method to only two rows:
```
if (!allow_empty_password && password.isEmpty())  
    throw new AuthenticationException("empty password is not supported");  
```
sorry, but i can't create public static boolean parseBoolean method in LdapAuthenticatorConfiguration class: 
it will produce error with not able to get properties variable into this method:
```
public static boolean parseBoolean(String propertyName, Boolean defaultValue) {
    String propertyValue = properties.getProperty(propertyName);
    if (propertyValue != null)
        return Boolean.parseBoolean(propertyValue);
    return defaultValue;
}
```
event when try to place this method to AbstractLDAPAuthenticator class, it will produce error - _"non-static variable properties cannot be referenced from a static context"_.